### PR TITLE
[Gutenberg] [Aztec] Prevent headings from becoming bold after words are autocorrected

### PR DESCRIPTION
**Pull request by twstokes**
This PR upgrades Aztec to version `1.19.8` and points to the Gutenberg Mobile commit that incorporates this change.

**Related:**
- https://github.com/wordpress-mobile/gutenberg-mobile/issues/4063
- https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1334
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/4557
- https://github.com/WordPress/gutenberg/pull/38633

### To test:

**Prerequisites:**
- A WordPress.com site running the [Seedlet theme](https://wordpress.com/theme/seedlet).

1. Create a new Post
2. Add a `Heading` block
3. In the block type: "Hello i'm testing a heading." (autocorrect should intervene on "i'm")
4. Observe the heading is now "Hello I'm testing a heading."
5. Observe that when highlighting "I'm", the bold formatting button isn't depressed / enabled.
6. Publish the Post.
7. Observe that "I'm" isn't shown in bold on the site. (you can further inspect the HTML source to ensure `<strong>` tags don't wrap the text)

## Regression Notes
1. Potential unintended areas of impact
- Any text block in the block editor
- Replacing text in text blocks
- Autocorrecting any text block

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Used testing steps above
- Tested various non-heading blocks

3. What automated tests I added (or what prevented me from doing so)
- [A unit test has been added to Aztec.](https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1334)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17844-2d3eeda</code>. IPA is available <a href='https://appcenter-filemanagement-distrib5ede6f06e.azureedge.net/43294098-bf86-46fb-8295-7ae6d399ac60/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=179%2FC0QjHuD6khh8%2B2sJZrMmMOMp8XHUq3%2FtiTfxN3I%3D&se=2022-02-04T20%3A08%3A13Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17844-2d3eeda</code>. IPA is available <a href='https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/37c6bb42-a79c-43fa-9966-85d4b5db2815/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=qFHE%2FUcfd7NU%2B9Lx%2FwMDkIq7mHn7ew0tGy0bXzVM4pQ%3D&se=2022-02-04T20%3A05%3A38Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Review by guarani — COMMENTED** [review]
Just leaving a review here. I tested this yesterday in the WPiOS app and it worked well, but when testing in Aztec I noticed a potential blocker: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1334#pullrequestreview-870943836.
Feel free to re-request a review when ready.

**Comment by twstokes** [comment]
I've pointed to the [latest Aztec commit](https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1334#issuecomment-1030142361) that hopefully mitigates the issue found.

**Review by guarani — APPROVED** [review]
Approved via https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1334#pullrequestreview-874778283.
I also tested here with the installable build and the testing steps work as expected and I didn't notice any regressions.

**Comment by twstokes** [comment]
Awaiting approvals on the Gutenberg Mobile side before cutting the new Aztec release. Once the release is cut we'll point to the tag.

**Comment by twstokes** [comment]
This branch was caught up with the 19.2 release branch so that it would use GB Mobile `v1.71.1`. The release has not yet been merged back into `trunk`.

**Comment by twstokes** [comment]
@fluiddot, since you helped out with the related [Gutenberg](https://github.com/WordPress/gutenberg/pull/38633) + [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4557) PRs, could you check this PR as well? I have the [Podfile pointed](https://github.com/wordpress-mobile/WordPress-iOS/blob/39f36e5ae92e93f23d44827ca5b169a9ad50be4a/Podfile#L173) to the merged Gutenberg Mobile commit, which if I understand correctly, may need to be tagged before this PR is merged into `trunk`. Thank you! 🙇

**Comment by twstokes** [comment]
⚠️ I just did a quick test of the [generated binary](https://github.com/wordpress-mobile/WordPress-iOS/pull/17844#issuecomment-1039688487) on this PR and it failed - the bold heading issue still appeared. I'm looking into it. 👀

File: RELEASE-NOTES.txt
Comment (fluiddot) [inline RELEASE-NOTES.txt:7]: If the Gutenberg Mobile ref is not being updated, which would include this fix, I'm wondering whether we should add this entry, wdyt?

Comment (twstokes) [reply RELEASE-NOTES.txt:7]: That's a good point @fluiddot - I'd leave it out if we were just updating `WordPress-Editor-iOS`. After our discussion, though, the plan will be to include the GB Mobile upgrade as well with this PR.

**Comment by twstokes** [comment]
> ⚠️ I just did a quick test of the [generated binary](https://github.com/wordpress-mobile/WordPress-iOS/pull/17844#issuecomment-1039688487) on this PR and it failed - the bold heading issue still appeared. I'm looking into it. 👀

This is working now that the GB Mobile bundle has been generated and we're pointing [to that commit](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4585). 👍 After the Gutenberg Mobile PR is merged and a tag is created, we'll update this PR.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I confirm that the issue can't be reproduced in the build `pr17844-d47c82c`.

**NOTE:** Totally unrelated to this PR but I noticed that the picker for switching the site doesn't work in the installable build. Not sure if it's a known issue otherwise we should open it.

**Comment by fluiddot** [comment]
As a side note, it would be great to address the warning from Peril ([reference](https://github.com/wordpress-mobile/WordPress-iOS/pull/17844#issuecomment-1026128800)) and assign a milestone before merging the PR.

Screenshot 2022-02-16 at 14 18 43

**Comment by twstokes** [comment]
> **NOTE:** Totally unrelated to this PR but I noticed that the picker for switching the site doesn't work in the installable build. Not sure if it's a known issue otherwise we should open it.

Thanks for bringing this up @fluiddot! I'm running the same build and haven't been able to reproduce that. 😬

**Comment by twstokes** [comment]
This has been pointed to the [GB Mobile tag](https://github.com/wordpress-mobile/gutenberg-mobile/releases/tag/v1.72.0-alpha2).

**Pull request merged**
